internal/events/handler.go

```diff
@@ -121,7 +124,11 @@ func (h *EventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Confi
 func (h *EventHandlerImpl) propagateUpsert(e *UpsertEvent) {
 	switch r := e.Resource.(type) {
 	case *v1beta1.GatewayClass:
-		h.cfg.Processor.CaptureUpsertChange(r)
+		if string(r.Spec.ControllerName) != h.cfg.ControllerName {
```

Comment:
if we don't handle this special case here, will https://github.com/nginxinc/nginx-kubernetes-gateway/blob/9c677a9b802267aee9b66d5919292f94f7a86ed2/internal/state/graph/gatewayclass.go#L40-L53 still handle the GatewayClass with a wrong controller name as expected (ignore it)?

if so, do we really need the special case?

Reply:
We won't build the graph with the old GC, but the old GC will still be in the store. This was an attempt to remove it from the store. Unless we don't care?

Reply:
> We won't build the graph with the old GC, but the old GC will still be in the store. This was an attempt to remove it from the store. Unless we don't care?

I think it is fine, since we're not gonna process it. This way the logic to pick the right resources remains in the graph package, with some optimizations about the number of handled events and watched resources enabled in the manager